cudapoa/src/application_parameters.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+* Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#include "application_parameters.hpp"
+#include <fstream>
+#include <getopt.h>
+#include <iostream>
+#include <string>
+
+#include <claraparabricks/genomeworks/io/fasta_parser.hpp>
+#include <claraparabricks/genomeworks/utils/signed_integer_utils.hpp>
+#include <claraparabricks/genomeworks/version.hpp>
+
+namespace claraparabricks
+{
+
+namespace genomeworks
+{
+
+namespace cudapoa
+{
+
+ApplicationParameters::ApplicationParameters(int argc, char* argv[])
+{
+    struct option options[] = {
+        {"input", required_argument, 0, 'i'},
+        {"mode", required_argument, 0, 'm'},
+        {"full-alignment", no_argument, 0, 'f'},
+        {"band-width", required_argument, 0, 'b'},
+        {"graph-output", required_argument, 0, 'g'},
+        {"max-windows", required_argument, 0, 'n'},
+        {"gpu-mem-alloc", required_argument, 0, 'r'},
+        {"version", no_argument, 0, 'v'},
+        {"help", no_argument, 0, 'h'},
+    };
+
+    std::string optstring = "i:m:fb:g:s:n:r:vh";
+
+    int32_t argument = 0;
+    while ((argument = getopt_long(argc, argv, optstring.c_str(), options, nullptr)) != -1)
+    {
+        switch (argument)
+        {
+        case 'i':
+            input_paths.push_back(std::string(optarg));
+            break;
+        case 'm':
+            consensus_mode = std::stoi(optarg);
+            break;
+        case 'f':
+            banded = false;
+            break;
+        case 'b':
+            band_width = std::stoi(optarg);
+            break;
+        case 'g':
+            graph_output_path = std::string(optarg);
+            break;
+        case 'n':
+            max_windows = std::stoi(optarg);
+            break;
+        case 'r':
+            gpu_mem_allocation = std::stod(optarg);
+            break;
+        case 'v':
+            print_version();
+        case 'h':
+            help(0);
+        default:
+            exit(1);
+        }
+    }
+
+    if (gpu_mem_allocation <= 0 || gpu_mem_allocation > 1.0)
+    {
+        std::cerr << "gpu-mem-alloc should be greater than 0 and less than 1.0" << std::endl;
+        exit(1);
+    }
+
+    if (consensus_mode < 0 || consensus_mode > 1)
+    {
+        std::cerr << "consensus_mode can only be 0 (consensus) or 1 (msa)" << std::endl;
```

Comment:
the naming of `consensus_mode` may imply MSA is a mode of consensus, while MSA and Consensus are two possible outputs of cudaPOA analysis, it can be changed to analysis or output-type.

Reply:
Done. Renamed to result